Clemente/handlers/app.handler.js

```diff
@@ -0,0 +1,199 @@
+    let id = req.params.id;
+    if (valid.id(id)) {
+      let partner = db.getPartner(id);
+      if (partner) {
+        console.log(`DELETE /partners/${id} OK`);
+        db.deletePartner(id);
+        res.status(200).json({ message: "Deleted." });
+      } else {
+        console.log(`DELETE /partners/${id} NOT FOUND`);
+        res.status(404).json({ message: "The partner doesn't exist" });
+      }
+    } else {
+      console.log(`DELETE /partners/${id} INVALID ID`);
+      res.status(400).json({ message: "Invalid id." });
+    }
+  },
+  ///////////////////////////////////////// BOOKS ////////////////////////////////////
+  listAllBooks: (req, res) => {
+    console.log(`GET /books/`);
+    res.status(200).json(db.getTable("books"));
+  },
+  getBook: (req, res) => {
+    let id = req.params.id;
+    if (valid.id(id)) {
+      let book = db.getBook(id);
+      console.log(`GET /books/${id}`);
+      if (book) {
+        res.status(200).json(book);
+      } else {
+        res.status(404).json({ message: "The book doesn't exist." });
+      }
+    } else {
+      console.log(`GET /books/${id} INVALID ID`);
+      res.status(400).json({ message: "Invalid id." });
+    }
+  },
+  addBook: (req, res) => {
+    let { title } = req.body;
+    let { author } = req.body;
+    let { id } = req.body;
+
+    if (valid.string(title) && valid.string(author) && valid.id(id)) {
+      if (db.getBook(id)) {
+        console.log(`POST /books/${id} +1`);
+        db.addBook(id);
+        res.status(200).json({ message: "Book added to inventory." });
+      } else {
+        console.log(`POST /books${id} NEW`);
+        db.addEntry("books", { title: title, author: author, inventory: 1, id: id });
+        res.status(201).json({ message: "Book added to collection." });
+      }
+    } else {
+      console.log(`POST /books/${id} INVALID DATA`);
+      res.status(400).json({ message: "Invalid data." });
+    }
+  },
+  deleteBook: (req, res) => {
+    let id = req.params.id;
+    let book = db.getBook(id);
+
+    if (book != null) {
+      if (book.inventory != 0) {
+        console.log(`DELETE /books/${id} OK`);
+        res.status(200).json({ message: "Deleted" });
+        db.deleteBook(id);
+      } else {
+        console.log(`DELETE /books/${id} EMPTY`);
+        res.status(404).json({ message: "The inventory of this book is empty." });
+      }
+    } else {
+      console.log(`DELETE /books/${id} NOT FOUND`);
+      res.status(404).json({ message: "The book doesn't exist" });
+    }
+  },
+  ///////////////////////////////////////// LOANS ////////////////////////////////////
+  listAllLoans: (req, res) => {
+    console.log(`GET /loans/`);
+    res.status(200).json(db.getTable("loans"));
+  },
+  lentBook: (req, res) => {
+    let { Bid } = req.body;
+    let { Pid } = req.body;
+    let book = db.getBook(Bid);
+    let partner = db.getPartner(Pid);
+    if (valid.id(Pid) && valid.id(Bid) && valid.string(book) && valid.string(partner)) {

```

Comment:
We could have many loans of the same book for the same partner, how can we identify a proper loan given that each loan has it loan date.

Reply:
To simplify things, a partner can lent each book once.